modules/internal/DefaultRectangular.chpl

```diff
@@ -1170,64 +1170,53 @@ module DefaultRectangular {
       } else {
         const mdPDLow = dom.dsiDim(mdParDim).low;
         const chunk = mdInd2Chunk(mdPDLow + followThis(mdParDim).low);
-        if mdPDLow + followThis(mdParDim).high <= mData(chunk).pdr.high {
-          //
-          // entire followThis is in one multi-data chunk
-          // TODO: is it ever otherwise?
-          //
+        if debugDataParMultiDData {
+          // this code assumes followThis spans but a single chunk
+          assert(mdPDLow + followThis(mdParDim).high <= mData(chunk).pdr.high);
```

Comment:
Would something like this be better protected by the ``boundsChecking`` param so that we can have frequent testing of this condition?